actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -32,5 +44,13 @@ def method_for_action(action_name)
         "default_render"
       end
     end
+
+    private
+      def no_templates?
+        # Check and see if any templates exist for the given action name, and prefixes.
+        # Variants are provided also as an argument, to check in case a template exists
+        # for a format not specifically declared in the `respond_to` block.
+        !template_exists_in_any_format?(action_name.to_s, _prefixes, { variants: request.variant })
```

Comment:
We need to pass in the variant, or else a test like:

```ruby
def test_implicit_render_template_unknown_format_response
    assert_raises ActionController::UnknownFormat do
      get :xml_action, format: 'json'
    end
  end
```

will fail.

Added some additional `AV` tests to check for variants.